stonesoup/measures/state.py

```diff
@@ -458,8 +458,43 @@ def __call__(self, state1, state2):
             else:
                 raise ValueError(f'The input sizes are not compatible '
                                  f'({len(state1.particles)} != {len(state2.particles)})')
+
+        elif isinstance(state1, GaussianState) and isinstance(state2, GaussianState):
+
+            state1 = copy.copy(state1)
+            state2 = copy.copy(state2)
+
+            if self.mapping is not None:
+                state1.state_vector = state1.state_vector[self.mapping, :]
+                state2.state_vector = state2.state_vector[self.mapping2, :]
+
+                rows = np.array(self.mapping, dtype=np.intp)
+                columns = np.array(self.mapping, dtype=np.intp)
+                state1.covar = state1.covar[rows[:, np.newaxis], columns]
+
+                rows2 = np.array(self.mapping2, dtype=np.intp)
+                columns2 = np.array(self.mapping2, dtype=np.intp)
+                state2.covar = state2.covar[rows2[:, np.newaxis], columns2]
+
+            if state1.ndim == state2.ndim:
+
+                log_term = np.log(np.linalg.det(state2.covar) / np.linalg.det(state1.covar))
+
+                n_dims = state1.ndim
+
+                trace_term = np.trace(np.linalg.inv(state2.covar)@state1.covar)
+
+                mahalanobis_term = np.transpose(state2.state_vector - state1.state_vector) @ \
+                    np.linalg.inv(state2.covar)@(state2.state_vector - state1.state_vector)
```

Comment:
Matrix inversion can be quite slow, so worth doing in once. May as well calculate difference once as well.

```suggestion
                inv_state2_covar = np.linalg.inv(state2.covar)
                trace_term = np.trace(inv_state2_covar@state1.covar)

                delta = state2.state_vector - state1.state_vector
                mahalanobis_term = delta.T @ inv_state2_covar @ delta
```